src/actions/actions.js

```diff
@@ -216,18 +216,33 @@ export function moveCursorToNextChunk() {
   };
 }
 
-export function evaluateNotebook() {
-  return (dispatch, getState) => {
+export const nonRunnableChunkType = ["md", "css", "raw", ""];
+
+const chunkIsRunnable = chunk =>
+  !nonRunnableChunkType.includes(chunk.chunkType);
+
+const chunkNotSkipped = chunk =>
+  !(
+    chunk.evalFlags.includes("skipRunAll") ||
+    chunk.evalFlags.includes("skiprunall")
+  );
+
+export function evaluateNotebookUsingQueue(evalQueueInstance) {
+  return () => (dispatch, getState) => {
     const { jsmdChunks, kernelState } = getState();
+
     if (kernelState !== "KERNEL_BUSY") dispatch(setKernelState("KERNEL_BUSY"));
+
     jsmdChunks.forEach(chunk => {
-      if (!["md", "css", "raw", ""].includes(chunk.chunkType)) {
-        evalQueue.evaluate(chunk, dispatch);
+      if (chunkIsRunnable(chunk) && chunkNotSkipped(chunk)) {
```

Comment:
There might be a way of doing this so we're not adding boilerplate to a function (we'll probably have more flags in the future) but I'm sure this comment will get YAGNI'd and we can always revisit down the line ;)